tools/depcheck/pkg/cmd/pin.go

```diff
@@ -0,0 +1,134 @@
+package cmd
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/origin/tools/depcheck/glide"
+)
+
+var pinImportsExample = `# output the contents of a glide.yaml file with new
+# imports pinned to a SHA from a given glide.lock file
+%[1]s pin --glide.lock=/path/to/glide.lock --glide.yaml=/path/to/glide.yaml
+`
+
+type PinImportsOpts struct {
+	LockFile *glide.LockFile
+	YamlFile *glide.YamlFile
+
+	DecodedYamlBytes []byte
+
+	Out    io.Writer
+	ErrOut io.Writer
+}
+
+type PinImportsFlags struct {
+	LockFileName string
+	YamlFileName string
+}
+
+func (o *PinImportsFlags) ToOptions(out, errout io.Writer) (*PinImportsOpts, error) {
+	if len(o.LockFileName) == 0 || len(o.YamlFileName) == 0 {
+		return nil, fmt.Errorf("both a --glide.lock path and a --glide.yaml path must be specified")
+	}
+
+	yamlFileBytes, err := ioutil.ReadFile(o.YamlFileName)
+	if err != nil {
+		return nil, err
+	}
+	lockFileBytes, err := ioutil.ReadFile(o.LockFileName)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(yamlFileBytes) == 0 {
+		return nil, fmt.Errorf("--glide.yaml path contained an empty file")
+	}
+	if len(lockFileBytes) == 0 {
+		return nil, fmt.Errorf("--glide.lock path contained an empty file")
+	}
+
+	lockfile := &glide.LockFile{}
+	yamlfile := &glide.YamlFile{}
+	if err := lockfile.Decode(lockFileBytes); err != nil {
+		return nil, err
+	}
+	if err := yamlfile.Decode(yamlFileBytes); err != nil {
+		return nil, err
+	}
+
+	return &PinImportsOpts{
+		LockFile: lockfile,
+		YamlFile: yamlfile,
+
+		DecodedYamlBytes: yamlFileBytes,
+
+		Out:    out,
+		ErrOut: errout,
+	}, nil
+}
+
+func NewCmdPinImports(parent string, out, errout io.Writer) *cobra.Command {
+	flags := &PinImportsFlags{}
+
+	cmd := &cobra.Command{
+		Use:   "pin --glide.lock=FOO.lock --glide.yaml=FOO.yaml",
+		Short: "Updates all non-forked dependencies in a glide.yaml file to use the SHAs specified in a glide.lock file.",
```

Comment:
Outputs a glide.yaml with all unspecified repos pinned to use SHAs specified in a glide.lock file.